### ISSUE TITLE
angular 11 throws net::ERR_SSL_PROTOCOL_ERROR

### ISSUE THREAD

**Issue by snebjorn** [angular/angular-cli#19403]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/cli or @angular/core

### Is this a regression?

Yes, the previous version in which this bug was not present was: 10.2.3

Console output in Angular 11:

![image](https://user-images.githubusercontent.com/1266245/99274343-86309380-282a-11eb-9d15-adf67b940a8b.png)

![image](https://user-images.githubusercontent.com/1266245/99274507-b6783200-282a-11eb-8fbb-2accbe6d39d2.png)

### Description

It was working in Angular 10.

Console output:

![image](https://user-images.githubusercontent.com/1266245/99273194-f9d1a100-2828-11eb-94e5-d260f4160c93.png)

![image](https://user-images.githubusercontent.com/1266245/99273273-15d54280-2829-11eb-9bf3-568566e5ca6a.png)

## 🔬 Minimal Reproduction
Clone https://github.com/snebjorn/dotnet-angular11

1. Run `ng serve` in ClientApp folder
2. Run `dotnet run` in the root folder
3. Open the browser at https://localhost:5001/
4. Observe the errors

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 11.0.1
Node: 12.16.1
OS: win32 x64

Angular: 11.0.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1001.2
@angular-devkit/build-angular   0.1100.1
@angular-devkit/core            10.1.2
@angular-devkit/schematics      11.0.1
@angular/cli                    11.0.1
@schematics/angular             11.0.1
@schematics/update              0.1100.1
rxjs                            6.6.3
typescript                      4.0.5
</code></pre>

**Anything else relevant?**

The frontend is accessed via https://localhost:5001 because it's "hosted" with ASP.NET Core using `spa.UseProxyToSpaDevelopmentServer("http://localhost:4200");`. See https://docs.microsoft.com/en-us/aspnet/core/client-side/spa/angular?view=aspnetcore-5.0&tabs=visual-studio

The self-signed localhost certificate is trusted. `dotnet dev-certs https -t`. See https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-5.0&tabs=visual-studio#trust-the-aspnet-core-https-development-certificate-on-windows-and-macos

**Comment by prabh-62** [comment]
Alan, adding `publicHost` option doesn't help. Scheme should be `http` not `https`

<img width="696" alt="logs_ws" src="https://user-images.githubusercontent.com/13006107/99297112-c9ccd280-2815-11eb-94cd-2e97f4983987.png">

**Comment by alan-agius4** [comment]
Let me get back to you in this tomorrow.

**Comment by snebjorn** [comment]
I stumpled upon this https://github.com/dotnet/aspnetcore/issues/27790 seems relevant :)

**Comment by jr01** [comment]
I experience the same issue, but the workaround  `ng serve --publicHost=localhost:44300` does work for me.

My setup
* `ng serve` with all default settings. 
* asp.net core SPA services proxy `builder.UseProxyToSpaDevelopmentServer("http://localhost:4200");` and asp.net Kestrel server runs on `https://localhost:44300`

In Angular 10 when navigating to `https://localhost:44300` Angular makes a call to `https://localhost:44300/sockjs-node/....` and live reload works.

In Angular 11 when navigating to `https://localhost:44300` Angular makes a call to `http://localhost:4200/sockjs-node/....` which gets blocked by the browser (CSP).

In Angular 11 vendor.js I notice:
```
socket(socketUrl, onSocketMessage);
/* WEBPACK VAR INJECTION */}.call(this, "?http://localhost:4200&sockPath=/sockjs-node"))
```

As mentioned with `ng serve --publicHost=localhost:44300` live reload works in Angular 11 and the vendor.js contains 
`.../* WEBPACK VAR INJECTION */}.call(this, "?http://localhost:44300&sockPath=/sockjs-node"))`

**Comment by erictor** [comment]
The workaround mentioned above by jr01 definitely stopped the ERR_SSL_PROTOCOL_ERROR for me on a .NET 5.0 Angular 11 app using Visual Studio 2019.  However for me it did not resolve the other issue I've been experiencing which is that Live Reload has stopped working after upgrading to Visual Studio 2019 v16.8.0 (and 16.8.1).   My testing seems to indicate the live reload issue I'm experiencing is about VS v16.8.0/1 rather than the version of Angular.  I tried an Angular 8 app and live reload worked on VS releases below 16.8.0 but not after.

**Comment by prabh-62** [comment]
I created https://github.com/angular/angular-cli/issues/19406 as a separate issue since there could be a different issue with dotnet AspNetCore template and I don't want to hijack this thread

**Comment by alan-agius4** [comment]
Hi all, 

Thanks all for the input and all the users who provided the reproductions. I diid look at both reproduction this morning (https://github.com/prabh-62/angular-reverse-proxy and https://github.com/snebjorn/dotnet-angular11)

I will be creating a PR for this shortly. As a workaround you can use `--public-host=0.0.0.0:0`

Thanks.
